python/paddle/fluid/tests/unittests/mkldnn/test_conv2d_mkldnn_op.py

```diff
@@ -15,44 +15,158 @@
+    return out
 
-class TestMKLDNNWithStride(TestWithStride):
-    def init_kernel_type(self):
-        self.use_mkldnn = True
-        self.data_format = "NCHW"
 
+class TestConv2dMKLDNNBiasResidualOp(TestConv2dOp):
+    def init_group(self):
+        self.groups = 1
 
-class TestMKLDNNWithGroup(TestWithGroup):
     def init_kernel_type(self):
-        self.use_mkldnn = True
         self.data_format = "NCHW"
-
-
-class TestMKLDNNWith1x1(TestWith1x1):
-    def init_kernel_type(self):
         self.use_mkldnn = True
-        self.data_format = "NCHW"
-
-
-class TestMKLDNNWithInput1x1Filter1x1(TestWithInput1x1Filter1x1):
-    def init_kernel_type(self):
-        self.use_mkldnn = True
-        self.data_format = "NCHW"
+        self._cpu_only = True
+
+    def init_test_case(self):
+        self.fuse_bias = False
+        self.bias_size = [6]
+        self.fuse_residual = False
+        self.input_residual_size = [2, 6, 3, 3]
+        self.fuse_relu_before_depthwise_conv = False
+        self.pad = [0, 0]
+        self.stride = [1, 1]
+        self.input_size = [2, 3, 5, 5]  # NCHW
+        assert np.mod(self.input_size[1], self.groups) == 0
+        f_c = self.input_size[1] // self.groups
+        self.filter_size = [6, f_c, 3, 3]
+
+    def setUp(self):
+        TestConv2dOp.setUp(self)
+        output = self.outputs['Output']
+        if hasattr(self, "fuse_bias"):
+            if self.fuse_bias and hasattr(self, "bias_size"):
+                bias = np.random.random(self.bias_size).astype(self.dtype)
+                output = conv2d_bias_forward_naive(output, bias)
+                output = output.astype(self.dtype)
+                self.inputs['Bias'] = OpTest.np_dtype_to_fluid_dtype(bias)
+
+        if hasattr(self, "fuse_residual"):
+            if self.fuse_residual and hasattr(self, "input_residual_size"):
+                input_residual = np.random.random(
+                    self.input_residual_size).astype(self.dtype)
+                output = conv_2d_residual_naive(output, input_residual)
+                output = output.astype(self.dtype)
+
+                self.attrs['fuse_residual_connection'] = self.fuse_residual
+                self.inputs['ResidualData'] = OpTest.np_dtype_to_fluid_dtype(
+                    input_residual)
+
+        self.outputs['Output'] = output
+
+
+class TestMKLDNNFuseReluBiasResidual(TestConv2dMKLDNNBiasResidualOp):
+    def init_test_case(self):
+        # TestConv2dMKLDNNBiasResidualOp.init_test_case(self)
+        self.fuse_bias = True
+        self.fuse_residual = True
+        self.fuse_relu_before_depthwise_conv = True
+        self.bias_size = [6]
+        self.input_residual_size = [2, 6, 5, 5]
+        self.pad = [1, 1]
+        self.stride = [1, 1]
+        self.input_size = [2, 3, 5, 5]  # NCHW
+        assert np.mod(self.input_size[1], self.groups) == 0
+        f_c = self.input_size[1] // self.groups
+        self.filter_size = [6, f_c, 3, 3]
```

Comment:
Consider avoiding the code repetition by calling the parent `init_test_case` method from `TestConv2dMKLDNNBiasResidualOp`